install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -219,8 +219,11 @@ data:
   skip-crd-creation: "true"
 {{- end }}
 
-{{- if or .Values.ingressController.enabled .Values.gatewayAPI.enabled (and (hasKey .Values "loadBalancer") (eq .Values.loadBalancer.l7.backend "envoy")) }}
+{{- if or .Values.envoyConfig.enabled .Values.ingressController.enabled .Values.gatewayAPI.enabled (and (hasKey .Values "loadBalancer") (eq .Values.loadBalancer.l7.backend "envoy")) }}
```

Comment:
Suggestion: evaluate `enable-envoy-config` once to a variable, then consume that when, e.g. determining whether or not to create the secrets bits.

Reply:
Could you elaborate on this suggestion? The way I see it is the variable would have the value `.Values.envoyConfig.enabled` and the condition in the secret creation would use it but not become any simpler. This is due to the secrets namespace creation having other constraints:
- it can be turned off even if envoyConfig is enabled
- it should not be created if the same namespace was already created for ingressController or gatewayAPI